src/serializers/config.rs

```diff
@@ -163,6 +199,174 @@ impl TimedeltaMode {
     }
 }
 
+impl TemporalMode {
+    pub fn datetime_to_json(self, py: Python, datetime: &Bound<'_, PyDateTime>) -> PyResult<PyObject> {
+        match self {
+            Self::Iso8601 => Ok(datetime_to_string(datetime)?.into_py_any(py)?),
+            Self::Seconds => Ok(datetime_to_seconds(datetime)?.into_py_any(py)?),
+            Self::Milliseconds => Ok(datetime_to_milliseconds(datetime)?.into_py_any(py)?),
+        }
+    }
+
+    pub fn date_to_json(self, py: Python, date: &Bound<'_, PyDate>) -> PyResult<PyObject> {
+        match self {
+            Self::Iso8601 => Ok(date_to_string(date)?.into_py_any(py)?),
+            Self::Seconds => Ok(date_to_seconds(date)?.into_py_any(py)?),
+            Self::Milliseconds => Ok(date_to_milliseconds(date)?.into_py_any(py)?),
```

Comment:
You can drop outer `Ok` from a lot of the functions here, e.g.:

```suggestion
            Self::Iso8601 => date_to_string(date)?.into_py_any(py),
            Self::Seconds => date_to_seconds(date)?.into_py_any(py),
            Self::Milliseconds => date_to_milliseconds(date)?.into_py_any(py),
```